include/LeMonADE/updater/moves/MoveAddMonomerBase.h

```diff
@@ -0,0 +1,154 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
+
+--------------------------------------------------------------------------------*/
+
+#ifndef LEMONADE_UPDATER_MOVES_MOVEADDMOVEBASE_H
+#define LEMONADE_UPDATER_MOVES_MOVEADDMOVEBASE_H
+
+#include <LeMonADE/updater/moves/MoveBase.h>
+#include <LeMonADE/utility/Vector3D.h>
+#include <LeMonADE/utility/RandomNumberGenerators.h>
+
+/*****************************************************************************/
+/**
+ * @file
+ *
+ * @class MoveAddMonomerBase
+ *
+ * @brief Base class for all add monomer bfm-moves. Specialized versions are MoveAddScMonomer and MoveAddBccMonomer.
+ *
+ * @details This class provides a base add monomer moves. The implementation
+ * details (i.e. MoveAddScMonomer, MoveAddBccMonomer, optimized versions,etc.) are given by the template parameter.
+ * This way, using the "Curiously Recurring Template Pattern (CRTP) to avoid virtual functions
+ * but still providing their functionality. A specialized local
+ * move type derived from this class must then be constructed in the following
+ * way: "class MySpecialMove:public MoveAddMonomerBase<MySpecialMove>".
+ * It must implement the functions init,check and apply. Calls to the corresponding
+ * functions in this base class are then redirected to the specialized implementation.
+ * See for an example the class MoveAddScMonomer. 
+ * Since this class simply serves as a common base, the functions apply(), check(), and init() don't do
+ * anything particular.
+ *
+ * @tparam <SpecializedMove> name of the specialized move.
+ *
+ * @todo Test for this Base class?
+ **/
+/*****************************************************************************/
+template <class SpecializedMove> 
+class MoveAddMonomerBase:public MoveBase
+{
+public:
+  //here come the functions that are implemented by the specialization
+  //! Reset the probability
+  template <class IngredientsType> void init(const IngredientsType& ingredients);
+  //! Check if the move is allowed by the system given as argument.
+  template <class IngredientsType> void check(const IngredientsType& ingredients);
+  //! Apply the move to the system given as argument
+  template <class IngredientsType> void apply(IngredientsType& ingredients);
+  
+  //! setter function for type of new monomer
+  void setType(int32_t t){type=t;}
+  //! setter function for position of new monomer taking a VectorInt3
+  void setPosition(VectorInt3 pos){position=pos;}
+  //! setter function for position of new monomer taking a triple of ints
+  void setPosition(int32_t x,int32_t y,int32_t z){position.setX(x);position.setY(y);position.setZ(z);}
+  //! getter function for the type of the new monomer
+  int32_t getType() const{return type;}
```

Comment:
This directly addresses `FeatureAttributes` and its monomer property `MonomerAttributeTag`. Please rename it by `getTag()` and `setTag(int32_t tag)` and `int32_t monomerTag` to avoid confusion, because I would expect `type` refers to something inherent property.